HHS/Head-Start-TTADP
# Edit goals and prevent losing objectives

## Description of change

* Prevent accidental deselects of goals from losing objectives that had been added
* Goals new to this report can be edited in place.

## How to test

1. Create a report with a goal and two objectives
2. Deselect the goal
3. Re-select the goal
4. Verify that the previously entered objectives are still present
5. Edit the goal title to fix a typo

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-87


## Checklists

### Every PR



### Production Deploy


### After merge/deploy
